README.md

```diff
@@ -1,9 +1,166 @@
+This repository contains a mock service to test clients, and a test data generator to create test data files for the mock service, or to be used in testing of the parsec service, or parsec clients.
+
 See [this issue](https://github.com/parallaxsecond/parsec/issues/350) which describes the
 proposal.
 
+# Prerequisites
+To run the utilities in this repository, you will need python3 and pip installing.
+
+To install package prerequisites, run the following from this folder:
+
+```bash
+pip install -r requirements.txt
+```
+# Usage
+## Run Mock Service
+**TBD**
+
+## Generate Test Data
+To generate test data from test specs, run 
+
+```
+python generator/generator.py
+```
+
+This will parse the test specs in generator/testspecs and create test data in testdata/.  
+
+# Test Data
+The generator/generator.py script creates test data files in the testdata folder.  These test data files are intended for use, either with the mock service supplied here, or in other mock services that may be developed for the parsec service or parsec clients.  The test data files consist of the [test spec](#test-spec-format) (useful for the writer of a unit test), and test data, which contains base 64 encodes strings for a parsec operation and a corresponding result.
+
+The overall test data file format is, therefore:
+
+```yaml
+spec:
+    # for format see below
+test_data:
+  operation: EKfAXh4AAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAA
+  result: EKfAXh4AAQAAAAAAAAAAAAAAAAAAAAIAAAAAAAEAAAAAAAAACAE=
+```
+## Expected Use Of Test Data
+This repository does not dictate how the test data should be used by parsec client developers, but it was developed with the following use case in mind:
+
+A client developer wants to test a parsec operation in their client.  
+- They select an appropriate test data file that fulfils the needs of their test (or creates a new one in this repository and uses that)
+- They write client code that should generate a message to match the operation section of the [test spec](#test-spec-format).
+- They configure a mock service (the one here or a custom one) to expect the data defined in the test_data.operation part of the test data file as well as the data to send if that operation is received and what to send if it is not.
+- They stimulate their client under test to send the operation message to the mock service.
+- The mock service will check to see if the operation message it received matches the expected value (this is an exact byte match).
+  - If the match was successful, then it will return the configured match response
+  - Otherwise, it will return the configured non-match response
+- The client under test will attempt to decode the response.  The results can be checked by the test code.
+- If the spec.response.parse_should_succeed field is set to true, then the test code *should* expect a successful parse of the response data.  Otherwise, it *should* expect the parsing to fail.
+
+
+## Writing Tests
```

Comment:
Looking at the files in this PR - not sure if I missed something from the discussion on the issue for the mock provider, but to me it doesn't seem useful to have both the `spec` and the `test` YAML files. Is it because if only the `spec` were used, running the tests would always involve `protoc`, whereas with `test` you only need Python?

Reply:
that was my logic for adding the test files to git - it removes that extra dependency from the user.  It also means, that if the client dev is using the test data files directly (without the mock service), which is a valid use case, they don't even need python.

Reply:
Alright, makes sense!